### PR TITLE
models: user affiliations schema disallow html attrs

### PR DESCRIPTION
**Pull request by zzacharo**
:heart: Thank you for your contribution!

### Description

Please describe briefly your pull request.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_accounts/profiles/dicts.py [hunk 1]
```diff
@@ -20,18 +20,19 @@ class ValidatedDict(dict):
...
+        """Validate the data with the dictionary's schema and return the valid value."""
         try:
             if self._schema is not None:
                 # without schema, we basically revert to a normal dictionary
                 # with more overhead
-                self._schema.load(data)
+                return self._schema.load(data)
         except ValidationError as error:
             raise ValueError(f"Validation failed: {error}")
+        return data
```
Comment (kpsherva) [inline invenio_accounts/profiles/dicts.py:35]: is it needed? the code should probably not reach here

Comment (zzacharo) [reply invenio_accounts/profiles/dicts.py:35]: if schema is None it acts as a normal dict...

Comment (kpsherva) [reply invenio_accounts/profiles/dicts.py:35]: in this case, minor for readability:
```suggestion
            def _validate(self, data):
        """Validate the data with the dictionary's schema and return the valid value."""
        if self._schema is None:
            return data
        try:
              # without schema, we basically revert to a normal dictionary
              # with more overhead
              return self._schema.load(data)
        except ValidationError as error:
            raise ValueError(f"Validation failed: {error}")
```

File: invenio_accounts/profiles/schemas.py [hunk 2]
```diff
@@ -43,7 +44,8 @@ class UserProfileSchema(Schema):
     """The default user profile schema."""
 
     full_name = fields.String()
-    affiliations = fields.String()
```
Comment (lnielsen) [inline invenio_accounts/profiles/schemas.py:46]: This is not correct. We use sanitised HTML when we want to store and output HTML, and string when we just want text.

**Comment by zzacharo** [comment]
closing

**Pull request closed without merging**